[Tests] GUI macOS `test_discovered_page` random fails

The random test fails during https://github.com/Tribler/tribler/pull/6629
[console](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_GUI_macOS/3744/console)


```python
=================================== FAILURES ===================================
_____________________________ test_discovered_page _____________________________

window = <tribler_gui.tribler_window.TriblerWindow object at 0x1321838b0>

    @pytest.mark.guitest
    def test_discovered_page(window):
        QTest.mouseClick(window.left_menu_button_discovered, Qt.LeftButton)
>       tst_channels_widget(window, window.discovered_page, "discovered_page", sort_column=2)

src/tribler-gui/tribler_gui/tests/test_gui.py:244: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

window = <tribler_gui.tribler_window.TriblerWindow object at 0x1321838b0>
widget = <tribler_gui.widgets.channelcontentswidget.ChannelContentsWidget object at 0x13380ac10>
widget_name = 'discovered_page', sort_column = 2, test_filter = True
test_subscribe = True

    def tst_channels_widget(window, widget, widget_name, sort_column=1, test_filter=True, test_subscribe=True):
        wait_for_list_populated(widget.content_table)
        screenshot(window, name=f"{widget_name}-page")
    
        # Sort
        widget.content_table.sortByColumn(sort_column, 1)
        wait_for_list_populated(widget.content_table)
        screenshot(window, name=f"{widget_name}-sorted")
        max_items = min(widget.content_table.model().channel_info["total"], 50)
        assert widget.content_table.verticalHeader().count() <= max_items
    
        # Filter
        if test_filter:
            old_num_items = widget.content_table.verticalHeader().count()
            QTest.keyClick(widget.channel_torrents_filter_input, 'r')
            wait_for_list_populated(widget.content_table)
            screenshot(window, name=f"{widget_name}-filtered")
            assert widget.content_table.verticalHeader().count() <= old_num_items
            QTest.keyPress(widget.channel_torrents_filter_input, Qt.Key_Backspace)
            wait_for_list_populated(widget.content_table)
    
        if test_subscribe:
            # Unsubscribe and subscribe again
            index = get_index_of_row_column(widget.content_table, 0, widget.model.column_position[Column.VOTES])
            widget.content_table.on_subscribe_control_clicked(index)
            QTest.qWait(200)
>           window.dialog.button_clicked.emit(0)
E           AttributeError: 'NoneType' object has no attribute 'button_clicked'

src/tribler-gui/tribler_gui/tests/test_gui.py:222: AttributeError
```

--- a/src/tribler-gui/tribler_gui/tests/test_gui.py
+++ b/src/tribler-gui/tribler_gui/tests/test_gui.py
@@ -215,14 +215,18 @@ def tst_channels_widget(window, widget, widget_name, sort_column=1, test_filter=
         wait_for_list_populated(widget.content_table)
 
     if test_subscribe:
-        # Unsubscribe and subscribe again
+        widget.content_table.sortByColumn(0, 0)
+        wait_for_list_populated(widget.content_table)
+        screenshot(window, name=f"{widget_name}-sorted-on-subscribe")
+        # Subscribe
         index = get_index_of_row_column(widget.content_table, 0, widget.model.column_position[Column.VOTES])
         widget.content_table.on_subscribe_control_clicked(index)
         QTest.qWait(200)
-        window.dialog.button_clicked.emit(0)
-        screenshot(window, name=f"{widget_name}-unsubscribed")
+
+        # Unsubscribe
         widget.content_table.on_subscribe_control_clicked(index)
         QTest.qWait(200)
+        screenshot(window, name=f"{widget_name}-unsubscribed")
         window.dialog.button_clicked.emit(0)
 
     # Test channel view

